### ISSUE TITLE
[Q] How to manage the folders for downloaded vs renamed files?

### ISSUE THREAD

**Issue by wsw70** [Radarr/Radarr#2128]
**Description:** I am having a hard time understanding how folders management works and what actions are to be expected from Radarr and which should be offloaded to other tools.

My folders setup is as follows:

- `/mnt/2TB1/downloading` hold the files which are currently downloaded by `deluge`
- when they are done, `deluge` moves them to `/mnt/2TB1/downloaded`
- from which I expected to use Radarr to rename them and move them to `/mnt/2TB1/Movies/<the renamed folder name>/<the rename file name>` 

Is this setup (the last bullet) possible with Radarr? Where should I configure the "downloaded but with an ugly name" and "renamed" folders? (respectively  `/mnt/2TB1/downloaded` and  `/mnt/2TB1/Movies` in my case)

**Radarr Version:** 0.2.0.852

**Comment by tynan-cr** [comment]
When you use the `Add a Movie` feature, the movies that show up in the results list have a field on the bottom of the panel for specifying the *destination* folder (renamed as you put it). If you set this location properly that is where the movies will go after post-processing.

As for how to configure the download handling (ugly naming as you put it), you should set this in your torrent client integration (`Download Client` settings panel). There is an option called `Completed Download Handling` which should be enabled.

You also have to connect Radarr with Deluge by adding a download client in this settings panel. Note: I don't use Deluge so I can't speak to whether or not Radarr will know about where the files have ended up after Deluge moves them but, I am pretty certain it will work fine.

Finally, you can go to the `Media Management` settings panel and choose how you want your files to be post-processed and renamed. Note: you can also enable the `Use Hardlinks instead of Copy` setting which will keep both the original file and the post-processed files in your filesystem, but will not use more space to do so.

To summarize, after setting this up, all 3 steps above should work:
1. File downloaded by client to a directory by Deluge.
2. Radarr handles the completed download by post-processing as specified.
3. File is copied, moved, or hardlinked to the directory you specified when adding the movie.

Let me know if you need more help!

**Comment by wsw70** [comment]
@tynan-cr : thank you for the detailed description. I use deluge for years and that part is fine. What I could not understand is how Radarr knows which movie, from the downloaded list, it should rename.

In retrospection, I guess that it keeps track of which movie was sent to deluge for download, its status is then monitored and finally (because it talks to deluge and understands where the completed movies are moved), the completed (ugly) version is renamed and hard-linked.

If this is indeed how it works then great, I got it :)

**Comment by galli-leo** [comment]
@wsw70 Radarr adds a label to those downloads it sends to deluge, so it will only import movies with that label (If you set the label to nothing in the Radarr settings it will try to import all downloads). So the movie does not necessarily have to be added by radarr, it just has to have the correct label in deluge.
